LayoutKit/Views/ReloadableViewUpdateManager.swift

```diff
@@ -0,0 +1,138 @@
+// Copyright 2016 LinkedIn Corp.
+// Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+
+import Foundation
+
+/// Delegate for a `ReloadableViewUpdateManager`.
+protocol ReloadableViewUpdateManagerDelegate: class {
+    weak var reloadableView: ReloadableView? { get }
+    var currentArrangement: [Section<[LayoutArrangement]>] { get set }
+}
+
+/// Manages updating a `ReloadableView` and its backing data.
+/// Updates may be applied incrementally, or in a batch.
+class ReloadableViewUpdateManager {
+    private weak var delegate: ReloadableViewUpdateManagerDelegate?
+    private weak var operation: NSOperation?
```

Comment:
You are correct, the protocol shouldn't care about whether this is weak or not. Removed.